client/src/nl/sensorlab/videowall/animation/baseanimations/liquidcolumns/LiquidColumn.java

```diff
@@ -0,0 +1,61 @@
+package nl.sensorlab.videowall.animation.baseanimations.liquidcolumns;
+import nl.sensorlab.videowall.animation.BaseAnimation;
+import processing.core.PApplet;
+import processing.core.PGraphics;
+import processing.core.PVector;
+
+public class LiquidColumn {
+
+	private LiquidColumnsAnimation parent; 
+
+	private PVector[] positions;
+	private float speed;
+
+	public LiquidColumn(LiquidColumnsAnimation p, float x, float speed) {
+		this.parent = p;
+		this.positions = new PVector[parent.PIXEL_RESOLUTION_Y];
+		this.speed = speed;
+		for(int y = 0; y < parent.PIXEL_RESOLUTION_Y; y++) {
+			this.positions[y] = new PVector(x, y);
+		}
+	}
+
+	public void update() {
+		// Ignore first node
+		for (int i = 1; i < positions.length; i++) {
+			// Update the y position
+			positions[i].y += speed;
+
+			// Get the distance
+			float distance = PVector.dist(parent.pushobject.position, positions[i]);
+
+			// Push nodes away
+			if(distance < parent.pushobject.radius){
+				PVector temp = parent.pushobject.position.copy();
+				temp.sub(positions[i]);
+				temp.normalize();
+				temp.mult(parent.pushobject.radius);
+				temp = PVector.sub(parent.pushobject.position, temp);
+				positions[i] = temp.copy();
+			}
+		}
+		for (int i = 1; i < positions.length; i++) {
+			PVector temp = PVector.sub(positions[i-1], positions[i]);
+			temp.normalize();
+			temp.mult(1);
+			positions[i] = PVector.sub(positions[i-1], temp);
+		}
+	}
+	
+	
+	public void draw(PGraphics g) {
```

Comment:
I think the draw and update method should be combined. As there's no separate physics and render loop, it has no benefits to separate these. You could have a separate update method that updates a single position, but one drawAnimationFrame call, now has three loops over all items of all positions in all liquidColums. That's a lot of loops.